taskiq/receiver/receiver.py

```diff
@@ -267,8 +267,16 @@ async def run_task(  # noqa: C901, PLR0912, PLR0915
             if timeout is not None:
                 if not is_coroutine:
                     logger.warning("Timeouts for sync tasks don't work in python well.")
-                target_future = asyncio.wait_for(target_future, float(timeout))
-            returned = await target_future
+
+                with anyio.fail_after(float(timeout)):
+                    while inspect.isawaitable(target_future):
```

Comment:
Why do we need this while loop? Won't it return an awaitable after first iteration? Can we just check it with if?

Reply:
done